src/ble.ts

```diff
@@ -6,10 +6,19 @@ export type ServiceOptions = {
   indicationUUID: string
 }
 
+export type BLEState = {
+  isRunning?: boolean
+  isScanning?: boolean
+  isAdvertising?: boolean
+  isConnected?: boolean
```

Comment:
These should all be defined with a default of false.

Reply:
Also, it would make more sense to just use hooks here.

Reply:
actually i dont really understand what you mean with using hooks here. can you please elaborate

Reply:
in peripheral and central i set the defaults. this is just the type. If you have anything else in mind please let me know.

Reply:
Agree we should have defaults here. That makes implementations that use it easier (as we now have 3 states: undetermined, false, true).

You can just define the initialState in the implementation right?

Reply:
hmm alright let me think about this. Not entirely sure what's best here

Reply:
@TimoGlastra

Reply:
not sure how to best do this either

Reply:
since this is not a component. I don't really understand what's meant with hooks here. Can we not just set the state by wrapping the getState() method in a useEffect in a component (for now)?

Reply:
state should be retrievable from that properly. Opinions @TimoGlastra?

Reply:
Sounds good to me!